README.md

```diff
@@ -118,6 +118,18 @@ The CLI may also be used without installation by using Docker.
 docker run --rm -v $(pwd):/data circleci/circleci-cli:alpine config validate /data/.circleci/config.yml --token $TOKEN
 ```
 
+## circleci-agent
+
+In order to maintain backwards compatibility with the `circleci` binary present in builds, some commands are proxied to a program called `circleci-agent`.
+
+This program must exist in your `$PATH` as is the case inside of a job.
```

Comment:
Do we want to/can we explain how to get `circleci-agent` into one's `$PATH`?

Reply:
Since `circleci-agent` is closed source you can't, it gets injected inside of jobs as part of the platform.

We used to CD a script that wraps a docker image which contains the agent (and is public), that was our original CLI before we built this one. However, we decided the agent should only be distributed where we execute builds.